app/plugin_api.go

```diff
@@ -57,6 +63,39 @@ func (api *PluginAPI) LoadPluginConfiguration(dest interface{}) error {
 	}
 }
 
+func (api *PluginAPI) ValidateConfig(config map[string]interface{}) bool {
+	// walk api down to manifest to see if settings meta data is defined
+	// plugin api infratructure might not be completly initialized
+	if config == nil || len(config) < 1 ||
+		api == nil ||
+		api.manifest == nil ||
+		api.manifest.SettingsSchema == nil ||
+		api.manifest.SettingsSchema.Settings == nil {
+		return false
+	}
+	ss := api.manifest.SettingsSchema.Settings
+	var result = true // assume we succeed
+	for _, s := range ss {
+		required := s.Required
+		if !required { // if required is undefined or false -> skip check for this key
+			// TODO considering loggin a warning if required is nil (undefined)
+			continue
+		}
+		k := s.Key
+		if required && config[k] == nil {
+			api.LogError("required plugin configugarion key missing",
```

Comment:
`configugarion` --> `configuration`

Reply:
1. yes, the 1. messages included in plugin_api.go only returned in the server logs.
2. (2) to help the admin resolve the error, for example the log could contain the following output:
`Error: for plugin [com.mattermost.sample-plugin] required key [someKey] is missing, please add configuration for key [someKey] within the following JSON path [Pluginsettings->Plugins->com.mattermost.sample-plugin->someKey] of type [bool] associated help text is [When true, an extra feature will be enabled!] within [config.json] `